FDB notify netlink notification on a mac move when software forwarding/VM

diff --git a/net/bridge/br_fdb.c b/net/bridge/br_fdb.c
index 5e93fc6..8b300e6 100644
--- a/net/bridge/br_fdb.c
+++ b/net/bridge/br_fdb.c
@@ -500,6 +500,7 @@ void br_fdb_update(struct net_bridge *br, struct net_bridge_port *source,
 {
 	struct hlist_head *head = &br->hash[br_mac_hash(addr, vid)];
 	struct net_bridge_fdb_entry *fdb;
+	struct net_bridge_port *old_src = NULL;
 
 	/* some users want to always flood. */
 	if (hold_time(br) == 0)
@@ -520,8 +521,11 @@ void br_fdb_update(struct net_bridge *br, struct net_bridge_port *source,
 					source->dev->name);
 		} else {
 			/* fastpath: update of existing entry */
+			old_src = fdb->dst;
 			fdb->dst = source;
 			fdb->updated = jiffies;
+			if (old_src != source)
+				fdb_notify(br, fdb, RTM_NEWNEIGH);
 		}
 	} else {
 		spin_lock(&br->hash_lock);
